src/ring_theory/ideal/local_ring.lean

```diff
@@ -284,6 +294,19 @@ end
 
 end
 
+lemma of_surjective [comm_semiring R] [local_ring R] [comm_semiring S] [nontrivial S]
+  (f : R →+* S) [is_local_ring_hom f] (hf : function.surjective f) :
```

Comment:
```suggestion
  {f : R →+* S} [is_local_ring_hom f] (hf : function.surjective f) :
```

Reply:
If this argument is implicit, the lemma doesn't work in the proof of `ring_equiv.local_ring` (In fact, the lemma `of_surjective` is only used in this proof). I guess this is because `function.surjective f` is actually `function.surjective ⇑f`. So I think the explicit argument is more useful than implicit one in this case.

Reply:
Hmm.. I wonder if we should spend some time in fixing this.

Maybe @eric-wieser has an idea of why this doesn't work?